rep-I0001.rst

```diff
@@ -156,27 +158,40 @@ The motion interface can be expressed as four variations:
 
-        topic_list:
-         - name: <topic name>
+        controller_joint_map:
+         - group: <controller group#>
            ns: <topic namespace>
-           group: <controller group#>
            joints:
             - <joint_1>
             - <joint_2>
             - <joint_N>
-         - name: <topic name>
+         - group: <controller group #>
            ns: ...
 
+This structure allows users to configure the robot interface to receive motion commands in several different ways:
+
+#. Single Topic
+    all groups use same namespace (e.g. `ns: ""`)
+#. Separate Topics
+    use group-specific namespaces
+#. Mixed Mode
+    joint map may contain multiple entries for the same group, in different namespaces.  This enables separate multi-arm topics for: left, right, both.
+  
+The interface node will always listen for motion commands on the same topic as the current interface (`joint_path_command`), inside the specified namespace for each group.
+
 State Interface
 =========
-The robot state interface encapsulates all the data coming FROM the robot controller, including joint position, velocity (if available), effort(if available), position error and general robot status information [#rbt_stat]_.  The implementation of the state interface is simpler than the motion interface because it can be generalized to the multi-arm case, where a single arm is just a specific example.
+The robot state interface encapsulates all the data coming FROM the robot controller, including joint position, velocity (if available), effort(if available), position error and general robot status information [#rbt_stat]_.
 
 The state interface is split into a joint state and robot status interface (although they will utilize the same socket connection, see `Communications Method`_).  The split allows joint state feedback to be sent at a higher rate than status information (which should change slowly).
- * Joint State - A single controller message is split into N JointState messages.
- * Robot Status - A single controller message that contains status information for each arm.
+
+As in the motion interface, users can specify which namespace to use for each robot group.  Unlike the motion interface, multiple groups that share a namespace are *always* combined into a single message (regardless of sync vs. async motion).  Separate namespaces can be used to receive separate messages for each motion group.
+
+For `JointState` and `JointTrajectoryFeedback` messages, aggregation of multiple robot groups is easily supported through the explicit joint-name list in the message.  For `RobotStatus` messages, the status of multiple groups is aggregated into a single message using simple logic and the `TriState` message type:  If all groups share the same status value, use that value, otherwise use `TriState::Unknown`.  Some fields (such as `in_motion`) may use modified logic (if "any" group in motion...).
```

Comment:
I'm struggling a bit with the rewording.  Perhaps I can explain my intent, and you can suggest more specific areas that could be clarified?

The intent is that "group" always refers to a Robot Motion Group, as defined on the robot controller, and always is directly/uniquely associated with a fixed set of physical motors.  The ROS client can use the "mapping rules" defined in the configuration parameter/YAML to publish the joint states on one or more topics ("namespaces").

If each Robot Motion Group is mapped to a separate namespace, the ROS interface looks much like the current single-group case: each topic has messages that contain only joint-state data for a single robot motion group.  For a dual-arm case, you might get a `/left/joint_states` topic with `[left_joint_1, ..., left_joint_6]` joints and a `/right/joint_states` topic with `[right_joint_1, ..., right_joint_6]` joints.

However, if multiple Robot Motion Groups are mapped to the same namespace, the multi-group core is more visible.  Rather than publish separate messages for each motion-group on the same topic, the intent is to combine joints from all groups into a single topic.  For a dual-arm robot, you might see: `/joint_states` with `[left_joint_1, ..., right_joint_6]`.

I was trying to highlight that the `RobotStatus` message is more complex, since it does not allow "lossless" aggregation of multiple `RobotStatus` messages into a single message.  The `DynamicGroupState` simple message defined above states that the robot will send a separate `RobotStatus` data-structure for each Motion Group.  My suggestion is that we can probably come up with some "common sense" logic rules for aggregation that should apply to the current `RobotStatus` fields.

Here's an example, for a dual-arm robot, with both Robot Motion Groups mapped to a single namespace:
  - from robot:
    - left: `drives_powered: Y, e_stopped: N, in_motion: Y, mode: AUTO`
    - right: `drives_powered: N, e_stopped: N, in_motion: N, mode: AUTO`
  - published to ROS:
    - status: `drives_powered: UNKNOWN, e_stopped: N, in_motion: UNKNOWN, mode: AUTO`
Alternatively, we could write the logic for some fields (e.g. `in_motion`) to return `TRUE` if *any* participating Robot Motion Groups were `TRUE` (instead of `UNKNOWN` in the mixed case).